code/Model/SiteTree.php

```diff
@@ -2755,6 +2764,21 @@ public function getControllerName() {
 	}
 
 	/**
+	 * Find the frontend controller name, which can be the current fully qualified class name
+	 * with Controller or _Controller added to it within the same namespace, or a customised
+	 * filename as per the $frontend_controller configuration option.
+	 *
+	 * @return string
+	 */
+	public function getFrontendControllerName()
+	{
+		if ($customController = $this->config()->frontend_controller) {
+			return (string) $customController;
+		}
+		return $this->getControllerName();
```

Comment:
Can't we do this the other way round, `getController` just does `Injector::inst()->create($this->getControllerName())`?

That seems more flexible to me, plus for templates don't we want the "controller name" and not the FQCN for the class the Injector instantiated?